wagtail/contrib/modeladmin/helpers/permission.py

```diff
@@ -44,7 +44,14 @@ def user_has_any_permissions(self, user):
         Return a boolean to indicate whether `user` has any model-wide
         permissions
         """
-        for perm in self.get_all_model_permissions().values('codename'):
+
+        # Cache the lookup as this is called often in list views
+        model_perms = getattr(self, '_model_perms_cache', None)
+        if model_perms is None:
+            model_perms = self.get_all_model_permissions().values('codename')
```

Comment:
It may be slightly cleaner to instead use `.values_list`, and then explicitly casting to a `list`. That way, you're only holding onto a list of strings, rather than a full queryset, reducing memory usage. The underlying query is the same, but the python-space stuff should be a bit cleaner. It would also then remove the need to pull out the `"codename"` key below.